File: benchmarks/accuracy/fixtures/go/interprocedural_sql_injection.go
package main

import (
	"database/sql"
	"fmt"
	"net/http"
)

var db *sql.DB

func findUser(name string) (*sql.Rows, error) {
	query := fmt.Sprintf("SELECT * FROM users WHERE name = '%s'", name)
	return db.Query(query)
}

func handler(w http.ResponseWriter, r *http.Request) {
	name := r.URL.Query().Get("name")
	findUser(name)
}

File: benchmarks/accuracy/fixtures/javascript/interprocedural_sql_injection.js
const express = require('express');
const db = require('./db');

function findUserByName(username) {
    const query = `SELECT * FROM users WHERE username = '${username}'`;
    return db.prepare(query).get();
}

function lookup(name) {
    return findUserByName(name);
}

exports.getUser = (req, res) => {
    const { username } = req.query;
    const user = lookup(username);
    return res.json(user);
};
